server/zanata-war/src/main/java/org/zanata/adapter/JsonAdapter.java

```diff
@@ -46,4 +48,9 @@ protected void updateParamsWithDefaults(IParameters params) {
         Parameters p = (Parameters) params;
         p.fromString("extractIsolatedStrings.b=true");
     }
+
+    @Override
+    protected String getIdFor(TextUnit tu, String content, String subDocName) {
+        return HashUtil.generateHash(tu.getName().concat(content));
```

Comment:
test-json-duplicateids.json won't merge as it requires similar key and content. The file test-json-duplicateentries.json however will (tested).
Looking at the data available in the tu at generating the ID, I don't even think I can provide the global id.